src/uris/deepLinks/deepLinkService.ts

```diff
@@ -312,6 +315,19 @@ export class DeepLinkService implements Disposable {
 		return repo.git.getBranch(targetId);
 	}
 
+	private async getPullRequest(targetId: string): Promise<PullRequest | undefined> {
+		const { repo, remote } = this._context;
+		if (!repo) return undefined;
+		const pullRequests = remote
+			? await this.container.integrations.getMyPullRequestsForRemotes(remote)
+			: await this.container.integrations.getMyPullRequests();
+
+		if (!pullRequests?.value) {
+			return;
+		}
+		return pullRequests.value.find(x => x.pullRequest.id === targetId)?.pullRequest;
+	}
+
```

Comment:
Since this is only for the deep link service, and since the PR ID lives in the context, I suggest just making this a `getPullRequestFromContext`, remove the `targetId` parameter, and just pull the `prId` directly from context as we are already doing for `repo` and `remote`. And if it is not defined, return `undefined` like we are already doing here for `repo`.

Furthermore, `getMyPullRequests` with no input parameters sounds like potentially a LOT of work for the small task of opening a PR's changes. Is there any reason we would arrive in this function without a defined `remote`?

Reply:
made it shorter, so I think there is no need to keep it as a separate method